test/e2e/index.test.ts

```diff
@@ -256,12 +256,13 @@ describe('e2e', () => {
   });
 
   describe('<TextField />', () => {
-    it('should handle `onClick` when clicking on the focused label position', async () => {
-      await renderFixture('TextField/TextFieldWithOnClick');
+    it('should fire `onClick` when clicking on the focused label position', async () => {
+      await renderFixture('TextField/OutlinedTextFieldOnClick');
 
       // execute the click on the focused label position
       await page.getByRole('textbox').click({ position: { x: 10, y: 10 } });
-      await page.waitForSelector('.MuiInputBase-root.Mui-error');
+      const errorSelector = page.locator('.MuiInputBase-root.Mui-error');
+      await errorSelector.waitFor();
```

Comment:
Nitpick: could you expand, why do you think that this approach is better than the one it replaces? 🤔 
IMHO, it looks somewhat pointless and way more verbose... 🤷

Reply:
I'm not sure we need this e2e test in the first place. It doesn't harm but I feel that the root problem is more <TextField onClick should be registered on the root element of the text field, we can't change that, as we did in https://github.com/mui/material-ui/pull/36892#discussion_r1293560474.

Reply:
> Nitpick: could you expand, why do you think that this approach is better than the one it replaces? 🤔 IMHO, it looks somewhat pointless and way more verbose... 🤷

@LukasTy I was looking up this stuff in the playwright docs, they say `waitForSelector` is discouraged so I was just trying out the alternative 😅

Reply:
> the origin of the problem is that <TextField onClick should be registered on the root element of the text field, we can't move it to a lower element

@oliviertassinari I've added another test for this case!

Reply:
> I was looking up this stuff in the playwright docs, they say `waitForSelector` is discouraged ([here](https://playwright.dev/docs/api/class-page#page-wait-for-selector)) so I was just trying out the alternative 😅

Oh. My bad, I must have mixed those up, because I remembered this suggestion the other way around... 🙈 
Anyway, I'd still prefer the API and cleanliness of the `waitForSelector` solution, but going with the recommended approach is always the safest choice. 👌